base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/pythoninterface/PythonInterface.java

```diff
@@ -107,13 +125,46 @@ public PythonInterface(String actionLoaderPythonScript) {
 				if (!pythonProcess.isAlive()) {
 					break;
 				}
-				LOG.error("> " + line);
+				if (line.equals("Python Ready")) {
+					handlePythonReadinessChange(true);
+					LOG.info("Python Ready");
+				} else {
+					LOG.error("> " + line);
+				}
 			}
 		} catch (IOException e) {
 			LOG.warn("Could not forward process errors");
 			LOG.warn(e);
 		}
 	};
+	
+	/**
+	 * The readiness of python has changed. 
+	 * Fire the property change and if python is no longer ready attempt to start python again.
+	 * 
+	 * @param ready Whether python is ready or not.
+	 */
+	protected void handlePythonReadinessChange(boolean ready) {
+		boolean wasPythonReady = pythonReady;
+		firePropertyChange(PYTHON_READINESS_PROPERTY, pythonReady, pythonReady = ready);
+		if (ready == false && wasPythonReady != ready) {
+			CompletableFuture.runAsync(() -> {
```

Comment:
As discussed, this is superfluous. Using just the Executor to submit the task will suffice.

Reply:
Yes, changed.